### PR TITLE
Stop poisoning function in debug mode

### PR DESCRIPTION
**Pull request by sophieyfang**
This is a cherry-pick of https://github.com/collectd/collectd/commit/fa36b45bfafec663e49639d40f4671e018e7ee10 (a fix for https://github.com/collectd/collectd/issues/2769, see https://github.com/collectd/collectd/pull/2804).

### CODE REVIEW

**Comment by qingling128** [comment]
Can we include the original PR that we are cherry picking? So that we have some context regarding to why this change is needed.

**Review by igorpeshansky — CHANGES_REQUESTED** [review]
I don't believe it's a good idea to cherry-pick this into the 5.5.2 branch. We will switch to the 5.8.1 branch at some point, which already contains this change. @qingling128, have we decided to stay with the 5.5.2 branch for long enough to justify this?

**Comment by qingling128** [comment]
@igorpeshansky - The reason we were considering this cherry pick is because this turned out to be the last blocker against Sles15 support for monitoring agent. And the content of the cherrypick did not seem harmful. As Sophie got so close to finishing this work (she shared a doc covering all the testing), wrapping it up saves us some context switch time and could benefit customers a bit earlier.

File: src/lua.c [hunk 1]
```diff
@@ -44,11 +44,6 @@
```
Comment (qingling128) [inline src/lua.c:40]: The original PR also removed the `DONT_POISON_SPRINTF_YET` definition. Should we do that as well?

image

**Review by qingling128 — COMMENTED** [review]
One comment regarding to `DONT_POISON_SPRINTF_YET`.

File: src/lua.c (hunk 1 above)
Comment (sophieyfang) [reply src/lua.c:40]: Nice Catch!

File: src/perl.c [hunk 2]
```diff
@@ -32,23 +32,11 @@
...
 
-#define DONT_POISON_SPRINTF_YET 1
-#include "collectd.h"
-
-#undef DONT_POISON_SPRINTF_YET
-
-#if HAVE_STDBOOL_H
-# include <stdbool.h>
-#endif
+#include <stdbool.h>
```
Comment (igorpeshansky) [inline src/perl.c]: This seems like an accidental change (merge conflict)... The 5.5.2 branch needs this `#if`/`#endif`.

File: src/utils_lua.c [hunk 3]
```diff
@@ -24,10 +24,7 @@
  *   Florian Forster <octo at collectd.org>
  **/
 
-/* <lua5.1/luaconf.h> defines a macro using "sprintf". Although not used here,
- * GCC will complain about the macro definition. */
-#define DONT_POISON_SPRINTF_YET
-
+#include "common.h"
```
Comment (igorpeshansky) [inline src/utils_lua.c]: Same here. `common.h` is already included 2 lines below. Let's remove this line.

**Review by igorpeshansky — CHANGES_REQUESTED** [review]
A couple of accidental changes from merge conflict resolution.

File: src/perl.c (hunk 2 above)
Comment (igorpeshansky) [reply src/perl.c]: [Optional] The original code had a space here. Might as well keep it consistent with the rest of the code.

**Review by igorpeshansky — CHANGES_REQUESTED** [review]
Minor.

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

**Review by qingling128 — APPROVED** [review]
LGTM

**Pull request merged**